Fix issues in release scripts

ankita240796 opened the pull request:
## Overview

1. This PR fixes or fixes part of N/A.
2. This PR does the following:

- Reverts ordering change for opening new tab in release scripts. Order is restored to open_new_tab -> print task. This is required since other way round was creating extra complications: Printing of task first normally expects a confirmation from user but without the url being shown, user does not know how to confirm. 
- Updates deploy branch to not run blocking bugs and last commit message check for hotfix branches.

## Essential Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The linter/Karma presubmit checks have passed locally on your machine.
- [x] "Allow edits from maintainers" is checked. (See [here](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for instructions on how to enable it.)
  - This lets reviewers restart your CircleCI tests for you.
- [x] The PR is made from a branch that's **not** called "develop".

## PR Pointers

- Oppiabot will notify you when you don't add a PR_CHANGELOG label. If you are unable to do so, please @-mention a code owner (who will be in the Reviewers list), or ask on [Gitter](https://gitter.im/oppia/oppia-chat).
- For what code owners will expect, see the [Code Owner's wiki page](https://github.com/oppia/oppia/wiki/Oppia%27s-code-owners-and-checks-to-be-carried-out-by-developers).
- Make sure your PR follows conventions in the [style guide](https://github.com/oppia/oppia/wiki/Coding-style-guide), otherwise this will lead to review delays
- Never force push. If you do, your PR will be closed.

ankita240796 committed:
Revert ordering change

ankita240796 committed:
Restrict checks to non hotfix branches

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-order

DubeySandeep commented on scripts/release_scripts/cut_release_or_hotfix_branch.py:
```diff
@@ -257,6 +257,10 @@ def execute_branch_cut(target_version, hotfix_number):
+    # the parent branch before checking out the new branch.
+    common.open_new_tab_in_browser_if_possible(
+        'https://travis-ci.com/oppia/oppia/branches')
     while True:
```
Any reason for this to be in a forever loop?

ankita240796 replied:
Checked, not required. Updated the code.

DubeySandeep commented on scripts/release_scripts/deploy.py:
```diff
@@ -493,19 +493,23 @@ def execute_deployment():
             check_travis_and_circleci_tests(current_branch_name)
 
             personal_access_token = common.get_personal_access_token()
             g = github.Github(personal_access_token)
```
Considering this value `g` is not required for hotfix deployment should we change this to:
```
personal_access_token = ...
if not hotfix_branch:
   last_commit_message = 
   if not last_commit_message.startswith:
       raise
   g = github.Github(personal_access_token) 
   repo = g.get_organization('oppia').get_repo('oppia')
   common.check_blocking_bug_issue_count(repo)
   common.check_prs_for_current_release_are_released(repo)

check_travis_and_circleci_tests(current_branch_name)
update_configs.main(personal_access_token)
...
```

[Trying to do all non hotfix related stuff at once and avoid multiple checks! (can you please check if this works?)]

ankita240796 replied:
Makes sense, updated and tested.

DubeySandeep reviewed (commented):
Thanks for the PR @ankita240796, all the changes LGTM! I've left a few minor question/suggestions PTAL!

Also, should we open the browser in the background using `<command> &` to fix the issue if the browser isn't open while running the script?

Make sure to test the task log as in my last release I found a few random logs after openeing browser, so I wanted to check that those logs don't ends up at the bottom!

ankita240796 committed:
Merge remote-tracking branch 'upstream/develop' into fix-order

ankita240796 committed:
Address review comments

ankita240796 commented:
> Thanks for the PR @ankita240796, all the changes LGTM! I've left a few minor question/suggestions PTAL!
> 
> Also, should we open the browser in the background using `<command> &` to fix the issue if the browser isn't open while running the script?
> 
> Make sure to test the task log as in my last release I found a few random logs after openeing browser, so I wanted to check that those logs don't ends up at the bottom!

Hi @DubeySandeep, I have addressed the review comments.

Regarding the issue with opening of browser while script is running, I am facing issues reproducing it. So, that will take some more time to test and fix, and I would handle that in a separate PR. Thanks!
